lib/media/streaming_engine.js

```diff
@@ -572,20 +584,34 @@ shaka.media.StreamingEngine = class {
           fullMimeType, this.manifest_.sequenceMode, stream.external);
     }
 
-    // Releases the segmentIndex of the old stream.
+    // Releases the segmentIndex of the old stream if safe to do so.
     // Do not close segment indexes we are prefetching.
     if (!this.audioPrefetchMap_.has(mediaState.stream)) {
       if (mediaState.stream.closeSegmentIndex) {
-        mediaState.stream.closeSegmentIndex();
+        if (this.config_.deferOldStreamCleanupDuringSwitch &&
```

Comment:
I agree with Alvaro... I'm not convinced that this should be optional, or that MSPs would want to control it.  I can see why you did this, since we generally do something like this in CAF so we can safely roll back changes that go wrong, but it doesn't fit as well in Shaka.

Reply:
Yeah, thinking of the rollback-friendly approach we often take in CAF was my initial thought as well. I'd expect this to benefit non-Cast platforms, and agree MSPs would not care to control this, so I've reverted the config.